### PR TITLE
tests: upgrade DM from 1.0 and 2.0 RC to 2.0 via TiUP

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

- add tests for upgrading DM from v1.0.6 to  2.0 nightly via TiUP
- add tests for upgrading DM from v2.0.0-rc and v2.0.0-rc.2 to 2.0 nightly via TiUP

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lance6716** [comment]
test failed at a TiDB bug https://github.com/pingcap/tidb/issues/20446 .
we might change testing text now

File: .github/workflows/upgrade-via-tiup.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+name: Upgrade via TiUP
+
+on:
+  pull_request:
+    branches:
+    - master
+    - release-2.0
```
Comment (csuzhangxc) [inline .github/workflows/upgrade-via-tiup.yml:7]: I'll remove these lines to avoid unnecessary running after received LGTM.

**Review by lance6716 — COMMENTED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: .github/workflows/upgrade-via-tiup.yml (hunk 1 above)
Comment (csuzhangxc) [reply .github/workflows/upgrade-via-tiup.yml]: removed

**Review by GMHDBJD — APPROVED** [review]

**Pull request merged**